### ISSUE TITLE
[Feature]: Rename X-axis label “Step” to `step` to match the SDK API

### ISSUE THREAD

**Issue by songyuc** [wandb/wandb#10896]
### Summary
In the W&B web UI, the default X-axis option is labeled **“Step”** (capital S).  
However, in the Python SDK and internal history, the corresponding field is named `step` / `_step`.

From an API user’s perspective, it would be clearer and more consistent if the UI label matched the API field name (i.e. **`step`** instead of **“Step”**), or at least made the mapping explicit.

---

### Background and current behavior

In the Python SDK, we log with:

```python
with wandb.init() as run:
    for i in range(10):
        run.log({"validation_loss": 1 / (i + 1)})
        # or explicitly:
        # run.log({"validation_loss": 1 / (i + 1)}, step=i)
````

According to the docs:

* The X-axis is described as `step`, which increments by 1 each time `wandb.Run.log()` is called.

* In the workspace / panel settings docs, the X-axis option is listed as:

  > **Step** – Increments each time you call `wandb.Run.log()`

* In the line plot docs, it’s also described as:

  > Step increments by default each time `wandb.Run.log()` is called…

So semantically, the UI “Step” clearly corresponds to the SDK’s `step` / internal `_step` field, and by default it increments on every `run.log()` call unless the user explicitly passes `step=` (or an integration like TensorBoard takes over the step semantics).

However, in the web UI, the X-axis label is shown as **“Step”** (capital S), not `step`. For users who are reading the API docs and looking at `run.log(..., step=...)`, this capitalization difference can be confusing:

* It’s not obvious at first glance that “Step” in the UI **is exactly the same thing** as the `step` argument / `_step` history field.
* It can look like “Step” is a separate conceptual metric, instead of the concrete field that the SDK exposes and that integrations override.

---

### Why this matters

For advanced / API-oriented users who:

* Explicitly pass `step=` to control the X-axis, or
* Rely on integrations like TensorBoard (`sync_tensorboard=True`), where step semantics are subtle (e.g. docs mention *“Setting the step argument in `run.log()` is turned off when syncing TensorBoard …”* ),

having a clear 1:1 mapping between:

* **SDK parameter / field:** `step`, `_step`
* **UI label:** “Step”

is important for understanding what the X-axis actually represents.

Right now, the capitalization difference (`step` vs “Step”) adds a small cognitive mismatch when trying to reason about:

* which field controls the X-axis,
* how `step=` interacts with integrations,
* and how to debug “wrong X-axis” issues.

---

### Minimal example

```python
import wandb
import time

run = wandb.init(project="step-label-test")

for i in range(5):
    loss = 1.0 / (i + 1)
    run.log({"train/loss": loss}, step=i)
    time.sleep(0.2)

run.finish()
```

In the UI:

* The time-series chart’s X-axis default label appears as **“Step”**.
* The values correspond to the `step` argument passed above (`0, 1, 2, 3, 4`).

---

### Environment

* OS: Linux
* Integrations: plain Python script; also reproducible when using integrations like Stable-Baselines3 + TensorBoard + `sync_tensorboard=True`.

---

### Final note

This is a small UX / naming improvement, but it would make the API ↔ UI mapping much clearer for users who read the docs and manipulate `step` explicitly (or debug step-related issues in RL / TensorBoard sync workflows).

Thanks for considering!

### Suggested Solution

**Rename the X-axis label from “Step” to `step`**

   * UI X-axis dropdown would show `step` (in backticks or code style), matching the API:

     * `run.log(..., step=...)`
     * internal `_step` in history
   * This makes the mapping obvious: **“this axis is the same `step` you pass in code.”**
